helsingborg-stad/app-mitt-helsingborg
# Chore/lock and doc

## Explain the changes you’ve made

Locked (and therefore also upgraded some) versions of some tools to match CI versions.

## Explain why these changes are made

To avoid potential CI problems with "it works on my machine".

## Explain your solution

Locked:

* Node (^16.13.1)*
* Yarn (^1.22.17)
* npm (^8.1.2)*
* Ruby (2.7.5)
* Fastlane (~> 2.204.3)
* CocoaPods (~> 1.11.2)
* XCode (~> 13.2, used by Fastlane)
* Bundler (2.3.10, locked in Gemfile.lock)

*Note: Node/npm shouldn't matter/be used, but they are locked for completeness.

*Note: I was unable to lock Rubygems - not sure how to do.

## How to test

Concrete example:

1. Checkout this branch
2. Verify that your local setup works as usual (or if it doesn't, that whatever local workarounds you have still work).
3. (optional) Test any Fastlane lane (like ios adhoc build) if you have it setup.

## Tested environments

- [] Storybook on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.

(Optional)

A good example:
Let's consider using a 3rd party authentication provider for this, to offload MFA and other considerations as they arise and as the privacy landscape evolves.
AWS Cognito is a good option, so is Firebase. I'm happy to start researching this path. Let's also consider breaking this out into its own service.
We can then re-use it or share the accounts with other apps in the future.

---

## Checklist for reviewers

Here are a few points that are useful for reviewers to check.

The short list of the (maybe) most important points are:

- Does this PR work as expected?
- Does the PR properly only concern a single feature/bug/thing?
- Is the code clear and concise?
- Will the changes impact existing functionality?
- Are there any linting errors (ESLint) that should be resolved?
- Do the automatic GitHub checks pass?

However it is good if a reviewer can be as thorough as possible,
within reasonable limits of course.

### Functionality

- Does this PR work as expected?
- If this is a new feature, does it add value, or is it a sign of feature-creep?
- Will the changes impact existing functionality?
- Does the changes create inconsistencies somewhere else?
- Does the code work on different platforms if applicable?
- How does the code handle if unexpected values are passed to functions etc.?

### Readability, syntax, formatting, style

- Is the code clear and concise?
- Has styling rules (Prettier) been applied to included files?
- Are functions/variables/etc. aptly named?

### Design

- Is the code properly planned and designed?
- Is code organized/separated in a logical fashion?
- Does the code follow clean code practices to a reasonable degree?
- Is the code self-documenting enough?
- Is the code in sync with existing patterns/technologies?
- Does the code reuse existing functionality where applicable?
- Is the code designed with reusability in mind?
- Does the code have hard-coded values that can be refactored?

### Patterns, idioms, tests, best practices

- Are there any linting errors (ESLint) that should be resolved?
- Is the code properly typed (TypeScript)?
- Does the code utilize language features to their full potential?

### Debugging, testing, configuring

- Is appropriate logging used?
- Have tests been added/modified to accommodate the changes?
- Is the feature sufficiently configurable in case tweaks are required in the future?

### Performance, reliability, scalability

- Is the code well optimized?
- Does the code scale?
- Does the code handle failure scenarios?

### Security

- Is there any obvious potential for exploits and security threats?
- Does code check proper authorization where applicable?

### Etiquettes

- Are commits atomic (does only one thing per commit)?
- Does the PR properly only concern a single feature/bug/thing?
- Are commit messages well-written?
- Do the automatic GitHub checks pass?
